src/riak_kv_qry_compiler.erl

```diff
@@ -125,10 +243,42 @@ occurs(F, FF) ->
       end,
       0, FF).
 
+-spec compile_inverdist_funcalls_to_orderby(
+        [invdist_fn_precompiled_spec()]) -> {[invdist_fn_compiled_spec()], [{error, term()}]}.
+%% convert successfully validated invdist funcalls (a list of them,
+%% to allow for percentile(x, 0.42), percentile(y, 0.24)) into a list
+%% of ORDER BYs, LIMITs and OFFSETs; or report validation errors if any.
+compile_inverdist_funcalls_to_orderby(InvDistFuns) ->
```

Comment:
Since we only report the first error anyway, and since a map would be easier to read than a fold, is there any reason we can't use a map with a throw on error?